src/controllers/connectionManager.ts

```diff
@@ -462,6 +477,27 @@ export default class ConnectionManager {
 		);
 	}
 
+	public async showInstructionTextAsWarning(profile: IConnectionInfo, reconnectAction: IReconnectAction): Promise<void> {
+		const instructionText = `${LocalizedConstants.msgPromptSSLCertificateValidationFailed}`;
+		await this.vscodeWrapper.showWarningMessageAdvanced(instructionText,
+			{ modal: false },
+			[
+				LocalizedConstants.enableTrustServerCertificate,
+				LocalizedConstants.readMore,
+				LocalizedConstants.cancel
+			])
+			.then(async (selection) => {
+				if (selection && selection === LocalizedConstants.enableTrustServerCertificate) {
+					profile.encrypt = 'Mandatory';
+					profile.trustServerCertificate = true;
+					await reconnectAction(profile);
+				} else if (selection && selection === LocalizedConstants.readMore) {
+					this.vscodeWrapper.openExternal(Constants.encryptionBlogLink);
+					this.showInstructionTextAsWarning(profile, reconnectAction);
+				}
+			});
```

Comment:
nit: cleanup (use async/await style whenever possible)
```suggestion
		const selection = await this.vscodeWrapper.showWarningMessageAdvanced(instructionText,
			{ modal: false },
			[
				LocalizedConstants.enableTrustServerCertificate,
				LocalizedConstants.readMore,
				LocalizedConstants.cancel
			]);
		if (selection === LocalizedConstants.enableTrustServerCertificate) {
			profile.encrypt = 'Mandatory';
			profile.trustServerCertificate = true;
			await reconnectAction(profile);
		} else if (selection === LocalizedConstants.readMore) {
			this.vscodeWrapper.openExternal(Constants.encryptionBlogLink);
			this.showInstructionTextAsWarning(profile, reconnectAction);
		}
```